### PR TITLE
contrib/prosody: new package (0.12.4)

### PR DESCRIPTION
**Pull request by ttyyls**

### CODE REVIEW

File: contrib/prosody/files/prosody [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# prosody system service
+
+type = scripted
```
Comment (q66) [inline contrib/prosody/files/prosody:3]: make a real process service

Comment (nekopsykose) [reply contrib/prosody/files/prosody:3]: i wonder if reloading will work with it via signal

File: contrib/lua5.4-sec/template.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+pkgname = "lua5.4-sec"
```
Comment (q66) [inline contrib/lua5.4-sec/template.py:1]: lua5.4-luasec

File: contrib/lua5.4-socket/template.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+pkgname = "lua5.4-socket"
```
Comment (q66) [inline contrib/lua5.4-socket/template.py:1]: lua5.4-luasocket

File: contrib/lua5.4-unbound/template.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
+pkgname = "lua5.4-unbound"
```
Comment (q66) [inline contrib/lua5.4-unbound/template.py:1]: lua5.4-luaunbound

File: contrib/lua5.4-filesystem/template.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
+pkgname = "lua5.4-filesystem"
```
Comment (q66) [inline contrib/lua5.4-filesystem/template.py:1]: lua5.4-luafilesystem

File: contrib/lua5.4-expat/template.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
+pkgname = "lua5.4-expat"
```
Comment (q66) [inline contrib/lua5.4-expat/template.py:1]: lua5.4-luaexpat

File: contrib/prosody/files/prosody [hunk 7]
```diff
@@ -0,0 +1,8 @@
+# prosody system service
+
+type = bgprocess
```
Comment (q66) [inline contrib/prosody/files/prosody:3]: ... just make a proper process service

Comment (ttyyls) [reply contrib/prosody/files/prosody:3]: it didn't work

Comment (q66) [reply contrib/prosody/files/prosody:3]: didn't work how

Comment (q66) [reply contrib/prosody/files/prosody:3]: just using `/usr/bin/prosody` as the command line should be completely enough

Comment (ttyyls) [reply contrib/prosody/files/prosody]: that's what I started with early and again.. for some reason it just doesn't work

File: contrib/lua5.4-luacliargs/template.py [hunk 8]
```diff
@@ -0,0 +1,16 @@
+pkgname = "lua5.4-luacliargs"
```
Comment (q66) [inline contrib/lua5.4-luacliargs/template.py:1]: lua5.4-lua_cliargs

File: contrib/lua5.4-luacliargs/template.py [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+pkgver = "3.0.2"
+pkgrel = 0
+pkgdesc = "CLI argument parsing module for lua"
```
Comment (q66) [inline contrib/lua5.4-luacliargs/template.py:4]: for Lua

File: contrib/lua5.4-luafilesystem/template.py [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+        "-shared",
+        "-fPIC",
+    ]
+
+
+def do_build(self):
+    from cbuild.util import compiler
+
+    cc = compiler.C(self)
+    cc.invoke(["src/lfs.c"], "lfs.so")
```
Comment (q66) [inline contrib/lua5.4-luafilesystem/template.py:25]: you can drop the `init_configure` and pass the args as `flags=` and `ldflags=` respectively here

File: contrib/lua5.4-luasay/template.py [hunk 11]
```diff
@@ -0,0 +1,14 @@
+pkgname = "lua5.4-luasay"
```
Comment (q66) [inline contrib/lua5.4-luasay/template.py:1]: lua5.4-say

File: contrib/lua5.4-luapenlight/template.py [hunk 12]
```diff
@@ -0,0 +1,17 @@
+pkgname = "lua5.4-luapenlight"
```
Comment (q66) [inline contrib/lua5.4-luapenlight/template.py:1]: lua5.4-penlight

File: contrib/lua5.4-luamediator/template.py [hunk 13]
```diff
@@ -0,0 +1,19 @@
+pkgname = "lua5.4-luamediator"
```
Comment (q66) [inline contrib/lua5.4-luamediator/template.py:1]: lua5.4-mediator_lua

File: contrib/lua5.4-luasystem/template.py [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+pkgdesc = "Platform independent system calls for Lua"
+maintainer = "ttyyls <contact@behri.org>"
+license = "MIT"
+url = "http://lunarmodules.github.io/luasystem"
+source = f"https://github.com/lunarmodules/luasystem/archive/refs/tags/v{pkgver}.tar.gz"
+sha256 = "d3686cbb8c370e71864a13dce8240bc54a1b3ca4ab9d108190c8d115dce7fba0"
+
+
+def init_configure(self):
+    self.tool_flags["CFLAGS"] += [
```
Comment (q66) [inline contrib/lua5.4-luasystem/template.py:14]: ditto here with the flags and/or makefile

File: contrib/lua5.4-luaterm/template.py [hunk 15]
```diff
@@ -0,0 +1,33 @@
+pkgname = "lua5.4-luaterm"
```
Comment (q66) [inline contrib/lua5.4-luaterm/template.py:1]: lua5.4-lua-term

File: contrib/lua5.4-luaterm/template.py [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+makedepends = ["lua5.4-devel"]
+pkgdesc = "Terminal operations for Lua"
+maintainer = "ttyyls <contact@behri.org>"
+license = "MIT"
+url = "https://github.com/hoelzro/lua-term"
+source = f"{url}/archive/refs/tags/{pkgver}.tar.gz"
+sha256 = "8ff94f390ea9d98c734699373ca3b0ce500d651b2ab1cb8d7d2336fc5b79cded"
+
+
+def init_configure(self):
```
Comment (q66) [inline contrib/lua5.4-luaterm/template.py:13]: ditto with makefile/flags

File: contrib/prosody/files/sysusers.conf [hunk 17]
```diff
@@ -0,0 +1,3 @@
+# Cretae prosody system user
+
+u prosody - "Prosody user" /var/lib/prosody
```
Comment (q66) [inline contrib/prosody/files/sysusers.conf:3]: lowercase prosody

File: contrib/lua5.4-luafilesystem/template.py (hunk 10 above)
Comment (ttyyls) [reply contrib/lua5.4-luafilesystem/template.py:25]: just the regular makefile stuff: overriding cflags, relying on gnu install behavior, gratuitoulsy using gmake only syntax, and at times broken lua search paths

Comment (ttyyls) [reply contrib/lua5.4-luafilesystem/template.py:25]: i think manually specifying the flags here is less likely to cause headaches and patches down the line

File: contrib/lua5.4-lua-term/template.py [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+
+def do_build(self):
+    from cbuild.util import compiler
+
+    cc = compiler.C(self)
+    cc.invoke(
+        ["core.c"],
+        "term.so",
+        flags=[f"-I{self.profile().sysroot / 'usr/include/lua5.4'}"],
+        ldflags=["-shared", "-fPIC"],
```
Comment (q66) [inline contrib/lua5.4-lua-term/template.py:21]: -fPIC goes in regular flags in general (ditto elsewhere)

**Pull request closed without merging**